charts/geoserver/v0.3.4/templates/deployment.yaml

```diff
@@ -0,0 +1,166 @@
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}
+  {{- with .Values.labels }}
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with .Values.annotations }}
+  annotations: {{ toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ template "geoserver.fullname" . }}
+      release: {{ .Release.Name | quote }}
+  template:
+    metadata:
+      name: {{ template "geoserver.fullname" . }}
+      labels:
+        app: {{ template "geoserver.fullname" . }}
+        chart: {{ template "geoserver.chart" . }}
+        release: {{ .Release.Name | quote }}
+        heritage: {{ .Release.Service | quote }}
+        {{- with .Values.podLabels }}
+        {{- tpl . $ | nindent 8 }}
+        {{- end }}
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+        checksum/secrets: {{ include (print $.Template.BasePath "/secrets.yaml") . | sha256sum }}
+        {{- with .Values.podAnnotations }}
+        {{- (tpl . $) | nindent 8 }}
+        {{- end }}
+    spec:
+      securityContext:
+        fsGroup: 1000
+      containers:
+        - name: {{ template "geoserver.fullname" . }}
+          image: {{ template "geoserver.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.resources }}
+          resources: {{- toYaml .Values.resources | nindent 12 }}
+          {{- end }}
+          env:
+            {{- if .Values.enableJSONP }}
+            - name: ENABLE_JSONP
+              value: {{ .Values.enableJSONP | quote }}
+            {{- end }}
+            {{- if .Values.geoserverDataDir }}
+            - name: GEOSERVER_DATA_DIR
+              value: {{ .Values.geoserverDataDir | quote }}
+            {{- end }}
+            {{- if .Values.geowebcacheCacheDir }}
+            - name: GEOWEBCACHE_CACHE_DIR
+              value: {{ .Values.geowebcacheCacheDir | quote }}
+            {{- end }}
+            {{- if .Values.geoserverContextRoot }}
+            - name: GEOSERVER_CONTEXT_ROOT
+              value: {{ .Values.geoserverContextRoot | quote }}
+            {{- end }}
+            - name: GEOSERVER_ADMIN_USER
+              valueFrom:
+                secretKeyRef:
+                  name: {{ template "geoserver.secretName" . }}
+                  key: geoserver-user
+            - name: GEOSERVER_ADMIN_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ template "geoserver.secretName" . }}
+                  key: geoserver-password
+            {{- with .Values.extraPodEnv }}
+            {{- tpl . $ | nindent 12 }}
+            {{- end }}
+          ports:
+            - name: http-geoserver
+              containerPort: 8080
+          {{- if .Values.probe }}
+            {{- with .Values.probe }}
+            {{- tpl . $ | nindent 10 }}
+            {{- end }}
+          {{- else }}
+          startupProbe:
```

Comment:
Do you know if this probe happens internally or using the externally exposed geoserver? If it's the latter then this can be affected by env `GEOSERVER_CONTEXT_ROOT`